modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2RoleClaimTestCase.java

```diff
@@ -171,8 +167,7 @@ public void testSendAuthorizedPostAfterRoleUpdate() throws Exception {
 
         String encodedIdToken = ((JSONObject) obj).get("id_token").toString().split("\\.")[1];
         Object idToken = JSONValue.parse(new String(Base64.decodeBase64(encodedIdToken)));
-        Object roles = ((JSONObject) idToken).get(OIDC_ROLES_CLAIM_URI);
-        Assert.assertTrue((((JSONArray) roles).contains(OAUTH_ROLE)), "Id token does not contain updated role claim");
+        Assert.assertNull(((JSONObject) idToken).get(OIDC_ROLES_CLAIM_URI), "Id token must not contain role claim");
```

Comment:
shall we add some more details. e.g. Id token must not contain role claim which is not available in the requested scopes